### ISSUE TITLE
Add end line + column parsing in links

### ISSUE THREAD

**Issue by Vidminas** [microsoft/vscode#78624]
The PR <https://github.com/microsoft/vscode/pull/24832> added support for line and column indicators in clickable links.

Adding parsing for end line and column markers would allow clicking on a link to open a file and highlight a range, which would be very useful for using external tool tasks without having to necessarily use problem matchers.

Currently clicking on a link, such as `/path/to/my/file.txt:5:56` now opens `file.txt` and positions the cursor at line 5, column 56.
The new feature would make clicking on links, such as `/path/to/my/file.txt:5:56->5:72` open `file.txt` and highlight the range from line 5, column 56 to line 5, column 72.

**Comment by Tyriar** [comment]
Is this a format output by any prominent tools?

**Comment by Vidminas** [comment]
1) In VS Code itself, problem matchers support parsing locations with end markers.

In `vscode/src/vs/workbench/contrib/tasks/common/problemMatcher.ts` the parsing is defined as:
```
private parseLocationInfo(value: string): Location | null {
    if (!value || !value.match(/(\d+|\d+,\d+|\d+,\d+,\d+,\d+)/)) {
	    return null;
    }
    let parts = value.split(',');
    let startLine = parseInt(parts[0]);
    let startColumn = parts.length > 1 ? parseInt(parts[1]) : undefined;
    if (parts.length > 3) {
	    return this.createLocation(startLine, startColumn, parseInt(parts[2]), parseInt(parts[3]));
    } else {
	    return this.createLocation(startLine, startColumn, undefined, undefined);
    }
}
```
This expects the location format to be like `5,56,5,72` or `(5,56,5,72)`

2) Again, in VS Code itself, the `Range` type provides a `toString()` implementation, in ` vscode/src/vs/editor/common/core/range.ts` as:
```
/**
 * Transform to a user presentable string representation.
 */
public toString(): string {
	return '[' + this.startLineNumber + ',' + this.startColumn + ' -> ' + this.endLineNumber + ',' + this.endColumn + ']';
}
```

This is the format [5,56->5,72] and is what I based my tool's output on, with the modification of ommiting square brackets and using colons instead of commas.

3) Finally, again within VS Code, the terminal window link handler currently does not support end markers, but it does support start markers in a multitude of formats. Defined in `vscode/src/vs/workbench/contrib/terminal/browser/terminalLinkHandler.ts` as:
```
const lineAndColumnClause = [
	'((\\S*)", line ((\\d+)( column (\\d+))?))', // "(file path)", line 45 [see #40468]
	'((\\S*)",((\\d+)(:(\\d+))?))', // "(file path)",45 [see #78205]
	'((\\S*) on line ((\\d+)(, column (\\d+))?))', // (file path) on line 8, column 13
	'((\\S*):line ((\\d+)(, column (\\d+))?))', // (file path):line 8, column 13
	'(([^\\s\\(\\)]*)(\\s?[\\(\\[](\\d+)(,\\s?(\\d+))?)[\\)\\]])', // (file path)(45), (file path) (45), (file path)(45,18), (file path) (45,18), (file path)(45, 18), (file path) (45, 18), also with []
	'(([^:\\s\\(\\)<>\'\"\\[\\]]*)(:(\\d+))?(:(\\d+))?)' // (file path):336, (file path):336:9
].join('|').replace(/ /g, `[${'\u00A0'} ]`);
```

So to summarise, I think that at the very least the formats `1, 2, 3, 4`, `(1, 2, 3, 4)`, `[1,2 -> 3,4]` should be supported.

But it would also make sense to support formats like `1:2:3:4`, `1:2->3:4`, `[1:2->3:4]`, `1,2->3,4`, `lines 5->12`, `line 5, column 8 to line 5 column 22` to properly match existing functionality with start markers (this list is not exhaustive and how the "wordy" output formats should be phrased is also up for discussion).

**Comment by Tyriar** [comment]
Closing as out of scope as the terminal is focused on CLI tools which do not generally use this format. FYI @alexr00
